### ISSUE TITLE
click on overlaying objects misfire sometimes

### ISSUE THREAD

**Issue by inetfuture** [pixijs/pixi.js#2551]
pixi.js version 3.0.9

There are two Containers named A and B, A sit on top of B, and A is smaller than B.

By my experiment, click on A will trigger `click` event both on A and B, in that order.

But sometimes, click on A only trigger `click` event on B, is there any know issues related to this? I haven't try to extract a reproduce example, I'll do it if you need, but first, I want to know if this ring some bells to someone?

I noticed that there is change related to 3.0.11: Fixed interaction issue with overlaying objects. what's that about?

**Comment by englercj** [comment]
Have you tried to reproduce in the latest version of 3.x?

**Comment by viperscape** [comment]
I am also having this issue, in 3.x
I made the root stage clickable and objects in front of it also clickable, both actions fire. I'm guessing this is because the event-emitter fires on both. I've tried setting zIndex, which may have been deprecated anyways, but also made sure the draw order is correct. How can I make sure only the first hit succeeds in a mousedown event?

**Comment by inetfuture** [comment]
> How can I make sure only the first hit succeeds in a mousedown event?

Your issue is not exactly same with mine, but FYI, I attach properties to `mouseData` in the top object's event handler, then in the bottom object's event handler I can tell if it's triggered from another object. I think the `mouseData` is reused across event handlers. But I'm not sure if this is the right thing to do. @englercj

**Comment by viperscape** [comment]
@inetfuture yes thank you. I realize what I was doing wrong-- this other issue tipped me off (https://github.com/pixijs/pixi.js/issues/2362)
Good luck, and I think your issue may be related to: https://github.com/pixijs/pixi.js/issues/2364
